waltz-ng/client/common/grid-utils.js

```diff
@@ -0,0 +1,42 @@
+/*
+ * Waltz - Enterprise Architecture
+ * Copyright (C) 2016, 2017 Waltz open source project
+ * See README.md for more information
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Creates a column def to render an entity link
+ *
+ * eg: usage: mkEntityLinkGridCell('Source', 'source', 'none')
+ *
+ * @param columnHeading column display name
+ * @param entityRefField field name in grid data that stores the entity ref for which the link needs to be rendered
+ * @param iconPlacement icon position, allowed values: left, right, none
+ * @returns {{field: *, displayName: *, cellTemplate: string}}
+ */
+export function mkEnumGridCell(columnHeading, entityRefField, showIcon = false) {
+    return {
+        field: entityRefField,
+        displayName: columnHeading,
+        cellTemplate: `
+            <div class="ui-grid-cell-contents">
+                <waltz-enum-value type="'AuthoritativenessRating'"
+                                  show-icon="${showIcon}"
+                                  key="COL_FIELD">
+                </waltz-enum-value>
+            </div>`
+    };
+}
```

Comment:
I've raised an issue to move other grid based helper functions into this file.

Reply:
There is also link-utils which has a couple of grid helper functions for entity-ref.